specification/riskiq/Defender.Easm/main.tsp

```diff
@@ -0,0 +1,1926 @@
+        subjectAlternativeNames?: string[];
+        issuerAlternativeNames?: string[];
+        sources?: Source[];
+        firstSeen?: utcDateTime;
+        lastSeen?: utcDateTime;
+        count?: int64;
+        invalidBefore?: utcDateTime;
+        keySize?: int32;
+        keyAlgorithm?: string;
+        subjectLocality?: string[];
+        subjectState?: string[];
+        subjectCountry?: string[];
+        issuerLocality?: string[];
+        issuerState?: string[];
+        issuerCountry?: string[];
+        subjectOrganizations?: string[];
+        subjectOrganizationalUnits?: string[];
+        issuerOrganizations?: string[];
+        issuerOrganizationalUnits?: string[];
+        version?: int32;
+        certificateAuthority?: boolean;
+        selfSigned?: boolean;
+        sigAlgOid?: string;
+        recent?: boolean;
+        validationType?: SslCertAssetValidationType;
+    }
+
+    @doc("SslServerConfig")
+    model SslServerConfig {
+        tlsVersions?: string[];
+        cipherSuites?: string[];
+        firstSeen?: utcDateTime;
+        lastSeen?: utcDateTime;
+        count?: int64;
+        sources?: Source[];
+    }
+
+    @doc("SubResourceIntegrityCheck")
+    model SubResourceIntegrityCheck {
+        violation?: boolean;
+        firstSeen?: utcDateTime;
+        lastSeen?: utcDateTime;
+        count?: int64;
+        causePageUrl?: string;
+        crawlGuid?: string;
+        pageGuid?: string;
+        resourceGuid?: string;
+        expectedHash?: string;
+    }
+
+    @doc("TaskPageResponse")
+    model TaskPageResponse {
+        @statusCode statusCode: 200;
+
+        @doc("The total number of items available in the full result set.")
+        totalElements?: int64;
+
+        @doc("The link to access the next page of results.  Not set if at the end of the result set.")
+        nextLink?: string;
+
+        @doc("The items in the current page of results.")
+        value?: TaskResponse[];
+    }
+
+    @doc("TaskResponse")
+    model TaskResponse {
+        @statusCode statusCode: 200;
+
+        @doc("The unique identifier of the task.")
+        id?: string;
+
+        @doc("The time the task started.")
+        startedAt?: utcDateTime;
+
+        @doc("The time the task completed.")
+        completedAt?: utcDateTime;
+
+        @doc("The last time the status of the task was updated.")
+        lastPolledAt?: utcDateTime;
+
+        @doc("The state the task is in.")
+        state?: TaskState;
+
+        @doc("The phase the task is in.")
+        phase?: TaskPhase;
+
+        @doc("The reason the task was moved into its current state, if the task wasn't completed.")
+        reason?: string;
+
+        @doc("Attributes unique to the task.  This differs by task type.")
+        metadata?: unknown;
+    }
+
+    @doc("ValidateResponse")
+    model ValidateResponse {
+        @statusCode statusCode: 200;
+        error?: ErrorDetail;
+    }
+
+    @doc("WebComponent")
+    model WebComponent {
```

Comment:
Azure core rules require all of the properties of each model to have documentation, you can do this with `@doc`  decorators, or documentation comments.

In any case, you should try to provide real documentation about what the property contains.  Just adding a comment that gives the name of the property if not sufficient. examples:

```tsp
/** The name of the web component, this name must be unique with the scope fo the application */
name: string
```
or

```tsp
@doc("The name of the web component, this name must be unique with the scope fo the application")
name: string
```